### DISCUSSION TITLE
Attachcommand limits to prevent accidental usage

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#7299] [Enhancement]
Players don't realise the tech behind a feature, they usually think custom items are a mod or something. 

If we have an attached command made like a do some fireworks, play a sound, and give an item, it works great. But it also works when we touch containers, like a hopper or a chest. 

It would be nice, to be honest, if we can have 'one time use' items stored in chests, without assuming that players will change item in hand first, before opening the chest. They've now accidentally used their unique one or two time use item in the wrong location or world. 

Could cmi attachcommand perhaps come with some config.yml toggles or something or built in code that prevents clicking on certain things to execute? 

Maybe even "require-air: true" (which could be false by default to respec the old situation)
Or something like "ignore-entities: true" (which prevents using it when they are pvp-ing or hitting an armorstand)
And "ignore-containers: true" (which will just open the dispener, hopper, anvil, chest, etc without using the attachcommand)

An extra enhancement request: "move-only-between-inventory: true" kinda like a soulbind item, so they can't drop special items into lava etc, but they can still open a chest and store it, or put it on an item frame or armourstand.

**Comment by CloudeLecaw** [comment]
i would also wished to see some kind of may a optional permission, that when its not set in a world, that players cant use the item to may waste a useage.
example: we have a stick that calls a rideable dragon.
when the player used the stick it takes one useage of the stick and the dragon is summoned. we blocked this summon of dragon in some worlds. (the useage of the command behind) but the useage of the item is still taken if the player tried to use the item in that world.  with a permission that the useage only is taken when player has permission, this issue because of missuseage could be prevented :)

**Comment by mrfloris** [comment]
I also just spotted this "placed" request; https://github.com/Zrips/CMI/discussions/7294

**Comment by Zrips** [comment]
Will make some adjustments with next update for it so that you can use something like 
`/cmi attachcommand !limiteduse:55!;;check:[interactedBlock]==null#! check:[interactedEntity]==null#! asConsole! cmi heal [playerName]` to get desired result. In this case it would only trigger while not clicking on entity or a block. Same thing can be used to limit item usage by permission access.

**Comment by mrfloris** [comment]
Thank you a lot for this quick turn-around. I am excited about this tweak.

**Reply by CloudeLecaw** [reply]
dito :D
